pkg/services/k8s/client/clientset.go

```diff
@@ -0,0 +1,127 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/grafana/grafana/pkg/kindsys/k8ssys"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8schema "k8s.io/apimachinery/pkg/runtime/schema"
+	memory "k8s.io/client-go/discovery/cached"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+)
+
+var (
+	// ErrCRDAlreadyRegistered is returned when trying to register a duplicate CRD.
+	ErrCRDAlreadyRegistered = errors.New("error registering duplicate CRD")
+)
+
+// Clientset is the clientset for Kubernetes APIs.
+// It provides functionality to talk to the APIs as well as register new API clients for CRDs.
+type Clientset struct {
+	config *rest.Config
+
+	clientset kubernetes.Interface
+	extset    apiextensionsclient.Interface
```

Comment:
i switched these to interfaces instead of pointers so that it might be easier to write tests